app/controllers/tasks.js

```diff
@@ -263,15 +277,71 @@ export default class TasksController extends Controller {
     }
   }
 
-  @action async handleUpdateTask(taskId, error) {
+  @action async handleUpdateTask(taskId, resetCurrentTask) {
+    this.resetCurrentTask = resetCurrentTask;
     const taskData = this.taskFields;
-    if (taskData.percentCompleted === TASK_PERCENTAGE.completedPercentage) {
+    const currentTask = this.getTaskById(taskId);
+    const isCurrentTaskStatusInProgress =
+      currentTask.status === this.TASK_KEYS.IN_PROGRESS;
+    const isCurrentTaskStatusBlock =
+      currentTask.status === this.TASK_KEYS.BLOCKED;
+    const isNewTaskStatusInProgress =
+      taskData.status === this.TASK_KEYS.IN_PROGRESS;
+    const isNewTaskStatusBlock = taskData.status === this.TASK_KEYS.BLOCKED;
+    const isCurrProgress100 =
+      parseInt(currentTask.percentCompleted || 0) === 100;
+    const isCurrProgress0 = parseInt(currentTask.percentCompleted || 0) === 0;
+
+    if (this.dev && taskData.status) {
+      if (
```

Comment:
Can't we somehow use switch case?
I'm seeing too many if statements

Reply:
> Can't we somehow use switch case? I'm seeing too many if statements If we can't then minimize the if conditions as suggested in other comments.

good suggestion

Reply:
Done,